lib/FrameManager.js

```diff
@@ -696,7 +696,8 @@ class Frame {
     const waitForVisible = !!options.visible;
     const waitForHidden = !!options.hidden;
     const polling = waitForVisible || waitForHidden ? 'raf' : 'mutation';
-    return this.waitForFunction(predicate, {timeout: options.timeout, polling}, selectorOrXPath, isXPath, waitForVisible, waitForHidden);
+    const timeout = helper.isNumber(options.timeout) ? options.timeout : 30000;
```

Comment:
is this a drive-by? Looks like an unrelated bugfix.

Reply:
Actually it is related to this comment:

https://github.com/GoogleChrome/puppeteer/pull/2084#comment-170343800

Reply:
Your comment link didn't work for me.

But if I understand correctly, the `page.waitForSelector`/`page.waitForXPath` don't currently have default timeout 30 seconds (despite documentation claiming so), and this change brings it back.

Reply:
ah I see now. Thank you for the explanation.